proxy/serve.go

```diff
@@ -28,23 +29,39 @@ var (
 	// mu guards servers which contains the list
 	// of running proxy servers.
 	mu      sync.Mutex
-	servers []Server
+	servers = make(map[string]Server)
 )
 
+func CloseProxy(address string) error {
+	if srv, ok := servers[address]; ok {
```

Comment:
Shouldn't we lock the whole `if` statement since we are doing read+write? 🤔

Reply:
Yes

Reply:
Done